src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -14,13 +18,62 @@
   "v3")
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(defn class-info-from-jruby->class-info-for-json
+  "Convert a class info map received from the jruby service into an
+  appropriate map for use in serializing an environment_classes response to
+  JSON"
+  [info-from-jruby environment]
```

Comment:
Also, re: schema perf, I still have yet to witness an occasion where that was any kind of bottleneck for us, so I'd prefer to err on the side of correctness and optimize later.  But in this instance my comments were mostly about readability of the fn sigs, so if you wanted to skip the `^:always-validate` part I could live with that.